kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/maps/MapMatchersTest.kt

```diff
@@ -269,6 +269,35 @@ class MapMatchersTest : WordSpec() {
           |
             """.trimMargin()
          }
+         "test assertion that a map contains extra keys and find similarities" {
+            val e = shouldThrow<AssertionError> {
+               mapOf(
+                  sweetGreenApple to 1
+               ) should containExactly(mapOf(sweetGreenPear to 1))
+            }
+            e.message shouldBe """
+            |
+            |Expected:
+            |  mapOf(Fruit(name=apple, color=green, taste=sweet) to 1)
+            |should be equal to:
+            |  mapOf(Fruit(name=pear, color=green, taste=sweet) to 1)
+            |but differs by:
+            |  missing keys:
+            |    Fruit(name=pear, color=green, taste=sweet)
+            |  extra keys:
+            |    Fruit(name=apple, color=green, taste=sweet)
+            |
+            |Possible matches for missing keys:
+            |
+            | expected: Fruit(name=pear, color=green, taste=sweet),
+            |  but was: Fruit(name=apple, color=green, taste=sweet),
+            |  similarity: 67.00%,
+            |  fields:
+            |  name expected: "pear", but was: "apple"
```

Comment:
```suggestion
            |  The following fields did not match:
            |     name: expected <"pear">, but was: <"apple">
```

Reply:
Increased indent for increased level of detail, and follow the `expected: <expected.print()>, but was <actual.print()>`

Reply:
> Increased indent for increased level of detail, and follow the `expected: <expected.print()>, but was <actual.print()>`

done